main.py

```diff
@@ -112,24 +112,23 @@ def produce_kafka_messages(bucket, job_id, fixture_data, args):
     s3_client = boto3.client("s3")
     for s3_key in fixture_data:
         logger.debug(f"Processing key: {s3_key}")
-        line_no = 0
-        for line in (
+        file_no = 0
+        for file in (
             s3_client.get_object(Bucket=bucket, Key=s3_key)["Body"]
             .read()
-            .splitlines()
         ):
-            line_no += 1
+            file_no += 1
             try:
-                data = json.loads(line)
+                data = json.loads(file)
```

Comment:
I think we need to get rid of this assumption that the file is going to be valid json at all; we were going to change the producer to just send any garbage straight out again, right? In which case I'd be tempted to do `data = s3_client.get_object(Bucket=bucket, Key=s3_key)["Body"].read()` on line 117 and just send `data` out via the `producer.send()` call.

Reply:
We've a separate story for that, which I Wass trying not to do yet :-)

Reply:
done though, as per the discussions.